prow/gerrit/adapter/trigger.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package adapter
```

Comment:
Why did we dupe all this?

Reply:
I mostly wanted to keep the plugin self contained. But I'm open to extracting the common logic to some place or exporting it directly from the trigger plugin. wdyt?

Reply:
Moving the shared code out of the `trigger` plugin would be the cleanest. Perhaps a subpackage of `pjutil`?
I'm not sure if this should live with the `CouldRun` and `ShouldRun` functions in `config`.

Reply:
A shared package under `pjutil` sounds nice to me

Reply:
done.